src/pymor/core/defaults.py

```diff
@@ -382,12 +388,28 @@ def set_defaults(defaults):
         Dictionary of default values. Keys are the full paths of the default
         values (see :func:`defaults`).
     """
+    from pymor.tools import mpi
+    if mpi._event_loop_running and mpi.rank0:
+        mpi.call(_set_defaults, defaults)
     try:
         _default_container.update(defaults, type='user')
     except KeyError as e:
         raise KeyError(f'Error setting defaults. Key {e} does not correspond to a default')
 
 
+def get_defaults(_user=True, _file=True, _code=True):
```

Comment:
Are you using the underscores because `file` is a builtin? For that case, the general Python convention would be to write `file_`. In any case, I am generally in favor of using builtin names for arguments. (The standard library does so as well.)

Can you add a bit of documentation to the method?